### DISCUSSION TITLE
Text in csv files is shown incorrectly on GitHub

### DISCUSSION THREAD

**Discussion by klefo0** [dfint/.github#13] [Q&A] [answered]
The format convert right, but the string translate output is wrong badly.
![image](https://github.com/dfint/autobuild/assets/70630854/ed0b4c88-fa11-4327-8d7a-5d61b3e71bfb)
![image](https://github.com/dfint/autobuild/assets/70630854/59d5c7f3-b1d5-47f6-91ed-47e25d75ae67)
![image](https://github.com/dfint/autobuild/assets/70630854/71b4a4dd-5065-4281-9872-188e1a562d3b)
![image](https://github.com/dfint/autobuild/assets/70630854/41cfee33-e1a3-4651-9b83-3689bcd6fd9b)
![image](https://github.com/dfint/autobuild/assets/70630854/86790ace-9ab7-4b5d-92f5-b7ea063c5ae6)

**Comment by insolor** [comment, marked as answer]
This is a normal result. Github shows readable text only for text in utf-8 encoding and in (I guess) cp1252 (Western European Windows encoding).

Csv files are encoded in different encodings (as configured in [config.yaml](https://github.com/dfint/autobuild/blob/main/config.yaml)), so don't expect them to be shown correctly in GitHub or Excel (for example).

**Comment by klefo0** [comment]
Even I downloaded csv file and open it in notepad or notepad++, it shows the csv file encode is in ANSI not in UTF-8. Try it yourself

**Comment by insolor** [comment]
@klefo0 ANSI is not an encoding, this is rather a familiy of encodings used in Windows (see https://en.wikipedia.org/wiki/ANSI_character_set). Exact ANSI variant used in your system (and used as ANSI by Notepad++ and other editors) depends on your Windows language and configuration.

In general csv files here (in this project) are encoded in DOS encodings, with some exceptions. For Ukrainian cp1251 encoding ([Windows 1251](https://en.wikipedia.org/wiki/Windows-1251), one of so called ANSI encodings) is used, and the csv file is shown in this encoding correctly. The same for Belarusian.

![image](https://github.com/dfint/autobuild/assets/2442833/b7e73232-a78d-4c9d-a2f4-603ce64c03ae)

![image](https://github.com/dfint/autobuild/assets/2442833/66d1675e-f472-4090-ace5-3b51cf1a253f)

For Vietnamese not standard encoding [VISCII](https://en.wikipedia.org/wiki/VISCII) is used, and I guess it's nearly impossible to find an editor which supports it (maybe there are some specific patches for editors for it).

But in general, if you open a csv file using encoding specified in config.yaml, you'll see correct text. For example, Spanish + cp850:

![image](https://github.com/dfint/autobuild/assets/2442833/dcb5e487-5f9d-49ac-99af-7dc22feb4e25)

Note: vscode editor (which I use) doesn't even try to guess an encoding of a file, I just manually click on UTF-8 text, click "Reopen with Encoding" and choose correct encoding.
